### ISSUE TITLE
Orphaned backups

### ISSUE THREAD

**Issue by Ramsdenj** [digint/btrbk#76]
For some reason all of my backups are showing up as orphaned, is this intended usage?

```
btrbk -t list backups                                                                                                                                            john@Butters
source_host  source_subvol      snapshot_subvol               status    target_host  target_subvol
-----------  -----------------  ----------------------------  --------  -----------  --------------------------------------
-            /mnt/Butters/ROOT  -                             orphaned  -            /mnt/ButterBackup/ROOT/ROOT.20160307
-            /mnt/Butters/ROOT  -                             orphaned  -            /mnt/ButterBackup/ROOT/ROOT.20160308
-            /mnt/Butters/ROOT  -                             orphaned  -            /mnt/ButterBackup/ROOT/ROOT.20160309
-            /mnt/Butters/ROOT  -                             orphaned  -            /mnt/ButterBackup/ROOT/ROOT.20160310
-            /mnt/Butters/ROOT  -                             orphaned  -            /mnt/ButterBackup/ROOT/ROOT.20160311
-            /mnt/Butters/ROOT  -                             orphaned  -            /mnt/ButterBackup/ROOT/ROOT.20160312
-            /mnt/Butters/ROOT  -                             orphaned  -            /mnt/ButterBackup/ROOT/ROOT.20160313
-            /mnt/Butters/ROOT  -                             orphaned  -            /mnt/ButterBackup/ROOT/ROOT.20160314
-            /mnt/Butters/ROOT  -                             orphaned  -            /mnt/ButterBackup/ROOT/ROOT.20160315
-            /mnt/Butters/ROOT  -                             orphaned  -            /mnt/ButterBackup/ROOT/ROOT.20160316
-            /mnt/Butters/ROOT  -                             orphaned  -            /mnt/ButterBackup/ROOT/ROOT.20160317
-            /mnt/Butters/ROOT  -                             orphaned  -            /mnt/ButterBackup/ROOT/ROOT.20160318
-            /mnt/Butters/ROOT  -                             orphaned  -            /mnt/ButterBackup/ROOT/ROOT.20160319
-            /mnt/Butters/ROOT  -                             orphaned  -            /mnt/ButterBackup/ROOT/ROOT.20160320
-            /mnt/Butters/ROOT  -                             orphaned  -            /mnt/ButterBackup/ROOT/ROOT.20160321
-            /mnt/Butters/ROOT  -                             orphaned  -            /mnt/ButterBackup/ROOT/ROOT.20160322
-            /mnt/Butters/ROOT  -                             orphaned  -            /mnt/ButterBackup/ROOT/ROOT.20160323
-            /mnt/Butters/ROOT  -                             orphaned  -            /mnt/ButterBackup/ROOT/ROOT.20160323_1
-            /mnt/Butters/ROOT  /mnt/Butters/ROOT.20160323_2  -         -            /mnt/ButterBackup/ROOT/ROOT.20160323_2
-            /mnt/Butters/home  -                             orphaned  -            /mnt/ButterBackup/home/home.20160307
-            /mnt/Butters/home  -                             orphaned  -            /mnt/ButterBackup/home/home.20160308
-            /mnt/Butters/home  -                             orphaned  -            /mnt/ButterBackup/home/home.20160309
-            /mnt/Butters/home  -                             orphaned  -            /mnt/ButterBackup/home/home.20160310
-            /mnt/Butters/home  -                             orphaned  -            /mnt/ButterBackup/home/home.20160311
-            /mnt/Butters/home  -                             orphaned  -            /mnt/ButterBackup/home/home.20160312
-            /mnt/Butters/home  -                             orphaned  -            /mnt/ButterBackup/home/home.20160313
-            /mnt/Butters/home  -                             orphaned  -            /mnt/ButterBackup/home/home.20160314
-            /mnt/Butters/home  -                             orphaned  -            /mnt/ButterBackup/home/home.20160315
-            /mnt/Butters/home  -                             orphaned  -            /mnt/ButterBackup/home/home.20160316
-            /mnt/Butters/home  -                             orphaned  -            /mnt/ButterBackup/home/home.20160317
-            /mnt/Butters/home  -                             orphaned  -            /mnt/ButterBackup/home/home.20160318
-            /mnt/Butters/home  -                             orphaned  -            /mnt/ButterBackup/home/home.20160319
-            /mnt/Butters/home  -                             orphaned  -            /mnt/ButterBackup/home/home.20160320
-            /mnt/Butters/home  -                             orphaned  -            /mnt/ButterBackup/home/home.20160321
-            /mnt/Butters/home  -                             orphaned  -            /mnt/ButterBackup/home/home.20160322
-            /mnt/Butters/home  -                             orphaned  -            /mnt/ButterBackup/home/home.20160323
-            /mnt/Butters/home  -                             orphaned  -            /mnt/ButterBackup/home/home.20160323_1
-            /mnt/Butters/home  /mnt/Butters/home.20160323_2  -         -            /mnt/ButterBackup/home/home.20160323_2
```

I couldn't find much information on exactly what the status orphaned means. I know with btrfs it means here are unreachable blocks,  but I'm not sure if that is ok as everything appears OK. 

This is what my configuration looks like,is there anything wrong with having the orphaned backups?

```
 snapshot_preserve_daily    0
 snapshot_preserve_weekly   0
 snapshot_preserve_monthly  0

target_preserve_daily      20
target_preserve_weekly     10
target_preserve_monthly    all


# Backup to external disk mounted on /mnt/ButterBackup
volume /mnt/Butters
  # no snapshot, done by snapper.
  snapshot_create  no

  volume /mnt/Butters
    subvolume ROOT
      target send-receive    /mnt/ButterBackup/ROOT

    subvolume home
      target send-receive    /mnt/ButterBackup/home
```

**Comment by digint** [comment]
This is intended usage. "orphaned" states that "the snapshot that was used for this backup is no longer available", or more technically speaking: "there is no subvolume having an uuid pointed to by the received_uuid of the backup subvolume".
This is a term btrbk uses for "btrbk list backups" and "btrbk stats", and has nothing in common with unreachable blocks of btrfs.

So for your setup (having snapshot_preserve = 0) it is completely correct that all except the latest backups are orphaned, as all except the latest snapshots are deleted after "btrbk run".

As of writing this, I'm not so sure if the orphaned flag is really a useful information, or just confusing people... I'll think of this and might drop it in the next version.

Here's an example on how it looks on a more complex setup:
```
--------------------------------------------------------------------------------
Statistics (btrbk command line client, version 0.22.2)

    Date:   Thu Mar 24 10:57:52 2016
    Config: /etc/btrbk/btrbk.conf
    Filter: group=local
--------------------------------------------------------------------------------
/mnt/btr_system/root_gentoo              25 snapshots
^-- /mnt/btr_ext/_btrbk/root_gentoo.*    15 backups   (5 orphaned)
^-- /mnt/btr_backup/root_gentoo.*       100 backups   (75 orphaned)
/mnt/btr_system/kvm                       1 snapshots (up-to-date)
^-- /mnt/btr_ext/_btrbk/kvm.*             1 backups
^-- /mnt/btr_backup/kvm.*                35 backups   (34 orphaned)
/mnt/btr_data/home                       26 snapshots
^-- /mnt/btr_backup/home.*               98 backups   (74 orphaned)
```
While the orphaned backups give me some hint on how many "long-preserved" backups are around, it does not seem really useful on second thought...

**Comment by Ramsdenj** [comment]
Wait, so does that mean I can't use any of the orphaned backups? Do I 
need the old snapshots? I would think not, right?

Or does it mean the backup is good, but just the original snapshot has 
been deleted? If that's true I wonder if some more descriptive 
information should be put in the man pages, or like you said it should 
be dropped .

**Comment by digint** [comment]
The "orphaned" backups are perfectly fine, just the original snapshot was deleted.

As long as the last snapshot/backup pair is still around (which is ensured by btrbk even if you set snapshot_preserve_*=0), you will still be able to make incremental backups from them. But as soon as ALL backups get orphaned (e.g. if you delete the latest snapshots manually), an incremental backup is no longer possible.

> If that's true I wonder if some more descriptive information should be put in the man pages, or like you said it should be dropped.

Agreed. Haven't decided yet if I drop it, but if not I will write the documentation needed.

**Comment by Ramsdenj** [comment]
Thanks for the response. btrbk is great by the way.

**Comment by digint** [comment]
Added short explanation of "orphaned" status in `btrbk stats`, and dropped display on `btrbk list backups` (in 12d435d42a590aa985336614393705c335dcf18e, merged in [btrbk-v0.23.1](/digint/btrbk/releases/tag/v0.23.1))
